### ISSUE TITLE
Describe the exact behavior of "pull rebase"

### ISSUE THREAD

**Issue by nakedible-p** [sociomantic/git-hub#135]
Please document what the "pull rebase" command actually does on the github level.

> Although probably the most outstanding feature (and the one that motivated the creation of this tool) is the pull rebase command, which is the rebasing version of the GitHub Merge (TM) button.

This doesn't really give a concrete description what the command does differently from standard github rebase:

- git checkout my-branch
- git pull --ff-only (if there are remote edits on the branch)
- git rebase master
- git push -f (or --force-with-lease)

**Comment by leandro-lucarella-sociomantic** [comment]
Please take a look at the [man page](https://github.com/sociomantic/git-hub/blob/master/man.rst). As said there, the steps are:

1. git stash
2. git fetch pullhead
3. git checkout -b tmp FETCH_HEAD
4. git pull --rebase pullbase
5. git push pullbase
6. git checkout oldhead
7. git branch -D tmp
8. git stash pop

Does this answers your question?

**Comment by nakedible-p** [comment]
Yes, finally I understand what the command does! I managed to miss the manual entirely, even though it is linked from the README.

Git is hard because different people have different intuitive understandings of how it works, so probably no single explanation is going to satisfy everybody. However, FWIW, here are my comments on the pull rebase documentation:

- It might be a good idea to mention that the command directly modifies the base branch, not the pull itself. For some reason I had no idea why "pull rebase" would touch anything but the pull itself.
- "Close a pull request identified by PULL by rebasing its base branch" The *base branch* is not rebased, instead the pull branch is rebased on top of the base branch and then the resulting branch is pushed to base branch.
- I think the most common thing people working with git have encountered is "fast forward" merges. What this essentially does is just normally rebase the branch and then "fast forward" merge it to base branch.
- In the command listing, pullbase and pullhead probably mean "origin pullbase" and "origin pullhead", which might be obvious but took a few moments to understand.

**Comment by leandro-lucarella-sociomantic** [comment]
If you could make your suggestions in the shape of a PR, that would be awesome :innocent:

**Comment by llucax** [comment]
Changing milestone to Future as it seems like nobody is fixing this soon and is not very high priority...
